apps/ejabberd/src/mod_last.erl

```diff
@@ -145,41 +143,36 @@ now_to_seconds({MegaSecs, Secs, _MicroSecs}) ->
 %%%
 %%% Serve queries about user last online
 %%%
--spec process_sm_iq(ejabberd:jid(), ejabberd:jid(), ejabberd:iq())
-        -> ejabberd:iq().
-process_sm_iq(From, To,
-    #iq{type = Type, sub_el = SubEl} = IQ) ->
-    case Type of
-        set ->
-            IQ#iq{type = error, sub_el = [SubEl, ?ERR_NOT_ALLOWED]};
-        get ->
-            User = To#jid.luser,
-            Server = To#jid.lserver,
-            {Subscription, _Groups} =
-                ejabberd_hooks:run_fold(roster_get_jid_info, Server,
-                    {none, []}, [User, Server, From]),
-                case (Subscription == both) or (Subscription == from) or
-                     (From#jid.luser == To#jid.luser) and
-                     (From#jid.lserver == To#jid.lserver) of
-                    true ->
-                        UserListRecord =
-                        ejabberd_hooks:run_fold(privacy_get_user_list, Server,
-                                                #userlist{}, [User, Server]),
-                        ?TEMPORARY,
-                        Acc = mongoose_acc:new(),
-                        Res = ejabberd_hooks:run_fold(privacy_check_packet,
-                                                      Server, Acc,
-                                                      [User, Server, UserListRecord,
-                                                       {To, From,
-                                                        #xmlel{name = <<"presence">>,
-                                                               attrs = [],
-                                                               children = []}},
-                                                       out]),
-                        make_response(IQ, SubEl, User, Server,
-                                      mongoose_acc:get(privacy_check, Res, allow));
-                    false ->
-                        IQ#iq{type = error, sub_el = [SubEl, ?ERR_FORBIDDEN]}
-                end
+-spec process_sm_iq(ejabberd:jid(), ejabberd:jid(), ejabberd:iq()) -> ejabberd:iq().
+process_sm_iq(_From, _To, #iq{type = set, sub_el = SubEl} = IQ) ->
+    IQ#iq{type = error, sub_el = [SubEl, ?ERR_NOT_ALLOWED]};
+process_sm_iq(From, To, #iq{type = get, sub_el = SubEl} = IQ) ->
+    User = To#jid.luser,
+    Server = To#jid.lserver,
+    {Subscription, _Groups} =
+    ejabberd_hooks:run_fold(roster_get_jid_info, Server,
+                            {none, []}, [User, Server, From]),
+    case (Subscription == both) or (Subscription == from) or
```

Comment:
To be honest, I'm not sure how should I read all these conditions to validate if this is correct or not. Would it be possible to make easier to understand by f.e introduction a function(s) or assigning results to some meaningful names?